flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RexNodeJsonSerializer.java

```diff
@@ -237,33 +247,38 @@ private void serialize(
             case INTERVAL_MINUTE:
             case INTERVAL_MINUTE_SECOND:
             case INTERVAL_SECOND:
-                gen.writeNumberField(FIELD_NAME_VALUE, ((BigDecimal) value).longValue());
+            case DATE:
+            case TIME:
+            case TIMESTAMP:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                gen.writeStringField(FIELD_NAME_VALUE, value.toString());
+                break;
+            case BINARY:
+            case VARBINARY:
+                gen.writeStringField(FIELD_NAME_VALUE, ((ByteString) value).toBase64String());
+                break;
+            case CHAR:
+            case VARCHAR:
+                gen.writeStringField(FIELD_NAME_VALUE, ((NlsString) value).getValue());
                 break;
             case SYMBOL:
                 final SerializableSymbol symbol = calciteToSerializable((Enum<?>) value);
+                gen.writeStringField(FIELD_NAME_SYMBOL, symbol.getKind());
                 gen.writeStringField(FIELD_NAME_VALUE, symbol.getValue());
-                gen.writeStringField(FIELD_NAME_CLASS, symbol.getKind());
                 break;
             case SARG:
-                serialize((Sarg<?>) value, elementTypeName, gen, serializerProvider);
-                break;
-            case ROW:
-            case MULTISET:
-                gen.writeFieldName(FIELD_NAME_VALUE);
-                gen.writeStartArray();
-                for (RexLiteral v : (FlatLists.ComparableList<RexLiteral>) value) {
-                    serialize(v, gen, serializerProvider);
-                }
-                gen.writeEndArray();
+                serializeSargValue((Sarg<?>) value, elementTypeName, gen, serializerProvider);
                 break;
             default:
-                // TODO support ARRAY, MAP
-                throw new TableException("Unknown value: " + value + ", type: " + literalTypeName);
+                throw new TableException(
```

Comment:
You replaced the TODO for `array` &  `map` and removed support for `ROW` and `MULTISET`.
Just double checking, don't have those as literals, right?

Reply:
Never used by Flink. The code generator has also no logic for them.